server/src/pages/shot/editor.js

```diff
@@ -270,14 +270,22 @@ exports.Editor = class Editor extends React.Component {
         let moveX = movement[0];
         moveX = moveX == "*" ? ["x1", "x2"] : [moveX];
         for (let moveDir of moveX) {
-          selectedPos[moveDir] = this.truncateX(resizeStartSelected[moveDir] + diffX);
+          if (resizeDirection == "move" && (selectedPos.left == 0 && resizeStartSelected.left + diffX < 0 || selectedPos.right == this.canvasWidth && resizeStartSelected.right + diffX > this.canvasWidth)) {
```

Comment:
This line is long and difficult to read.  Perhaps assign the conditions into variables so they are named, so it reads `if (someThing && someOtherThing && oneMoreThing)`.

Also, this `if` does not need to be inside the `for` loop.